app/views/questionnaire.py

```diff
@@ -465,6 +483,68 @@ def perform_list_action(
     return list_collector_url
 
 
+def handle_primary_person_action(
```

Comment:
This is kind of a lot of args to this method. I think a couple (router/routing_path) could be removed/replaced if rather than return the next url it was determined following the questionnaire store update?

Reply:
The problem with this is that you end up having a load of if statements for the questionnaire store update, and then a load more ifs when you come to work out the next URL, since each of these block types have slightly different behaviour for both of those.